ob1k-concurrent/src/main/java/com/outbrain/ob1k/concurrent/ComposableFuture.java

```diff
@@ -119,6 +119,16 @@
   ComposableFuture<T> andThen(Consumer<? super T> consumer);
 
   /**
+   * On success of the current future, applies the side-effecting function to its result.
+   * In case of an error the error is continued forward.
+   * Returns a new future with the same value.
+   *
+   * @param consumer the result consumer for the current future result
+   * @return a future with same value
+   */
+  ComposableFuture<T> peek(Consumer<? super T> consumer);
```

Comment:
the reason I haven't added such method, is becase #andThen method provides Try object - on which on top you can perform same transformation functions (map, recover, etc).
I don't think there's a meaning in such method, and if so, should be j.u.f.Consumer of T and not Try<T> :)